test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -152,6 +166,135 @@ var _ = ginkgo.Describe("SchedulerWithWaitForPodsReady", func() {
 			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodClusterQ.Name, prodWl)
 			util.ExpectWorkloadsToBeWaiting(ctx, k8sClient, devWl)
 		})
+
+	})
+
+	var _ = ginkgo.Context("Short PodsReady timeout", func() {
+
+		ginkgo.BeforeEach(func() {
+			podsReadyTimeout = 3 * time.Second
+		})
+
+		ginkgo.It("Should requeue a workload which exceeded the timeout to reach PodsReady=True", func() {
+			const lowPrio, highPrio = 0, 100
+
+			ginkgo.By("create the 'prod1' workload")
+			prodWl1 := testing.MakeWorkload("prod1", ns.Name).Queue(prodQueue.Name).Priority(highPrio).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodWl1)).Should(gomega.Succeed())
+
+			ginkgo.By("create the 'prod2' workload")
+			prodWl2 := testing.MakeWorkload("prod2", ns.Name).Queue(prodQueue.Name).Priority(lowPrio).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodWl2)).Should(gomega.Succeed())
+
+			ginkgo.By("checking the 'prod1' workload is admitted and the 'prod2' workload is waiting")
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodClusterQ.Name, prodWl1)
+			util.ExpectWorkloadsToBeWaiting(ctx, k8sClient, prodWl2)
+
+			ginkgo.By("awaiting for the Admitted=True condition to be added to 'prod1")
+			gomega.Eventually(func() bool {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(prodWl1), prodWl1)).Should(gomega.Succeed())
+				return apimeta.IsStatusConditionTrue(prodWl1.Status.Conditions, kueue.WorkloadAdmitted)
+			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
```

Comment:
is it possible that prod1 admission get cancelled by the time we do this test?

Reply:
for example the test process gets evicted and the 3 seconds passes before we get a chance to do this check

Reply:
Unlikely, because adding the condition is the next thing workload_controller would do in the `Reconcile` function after `admission!=nil`. Also, parking the other workload as `waiting` is the next thing the scheduler would do.

Still, it is possible if the testing infra is under load, but hasn't happened in my testing, I guess I could bump the timeout to 4s or 5s proactively, but maybe we bump once there is some evidence this is needed. Just now I tested it in a loop 30 times with `stress --cpu ${number of my cores}` and no flakes.

Reply:
Pushed adding the comment. 

> Did you test with a smaller timeout?

Just now did 15 passes with the timeout=1s and all passed, all with the same stress. Got failing 1 out of 2 with 200ms.